src/javax/sip/TransactionState.java

```diff
@@ -214,286 +214,47 @@
  * @version 1.2
 
  */
-
-public final class TransactionState implements Serializable{
-
-
-
-    /**
-
-     * Constructor for the TransactionState
-
-     *
-
-     * @param transactionState  The integer value for the TransactionState
-
-     */
-
-    private TransactionState(int transactionState) {
-
-        m_transactionState = transactionState;
-
-        m_transStateArray[m_transactionState] = this;
-
-    }
-
-
-
-    /**
-
-     * This method returns the object value of the TransactionState
-
-     *
-
-     * @return  The TransactionState Object
-
-     * @param transactionState The integer value of the TransactionState
-
-     */
-
-    public static TransactionState getObject(int transactionState){
-
-        if (transactionState >= 0 && transactionState < m_size) {
-
-            return m_transStateArray[transactionState];
-
-        } else {
-
-            throw new IllegalArgumentException("Invalid transactionState value");
-
-        }
-
-    }
-
-
-
-    /**
-
-     * This method returns the integer value of the TransactionState
-
-     *
-
-     * @return The integer value of the TransactionState
-
-     */
-
-    public int getValue() {
-
-        return m_transactionState;
-
-    }
-
-
-
-    /**
-
-     * Returns the designated type as an alternative object to be used when
-
-     * writing an object to a stream.
-
-     *
-
-     * This method would be used when for example serializing TransactionState.EARLY
-
-     * and deserializing it afterwards results again in TransactionState.EARLY.
-
-     * If you do not implement readResolve(), you would not get
-
-     * TransactionState.EARLY but an instance with similar content.
-
-     *
-
-     * @return the TransactionState
-
-     * @exception ObjectStreamException
-
-     */
-
-    private Object readResolve() throws ObjectStreamException {
-
-        return m_transStateArray[m_transactionState];
-
-    }
-
+public enum TransactionState{
+    NULL_STATE(-1),
+    // This constant value indicates the internal value of the "Calling" 
+    CALLING(0),
+    // This constant value indicates the internal value of the "Trying" 
+    TRYING(1),
+    // This constant value indicates the internal value of the "Proceeding" 
+    PROCEEDING(2),
+    // This constant value indicates the internal value of the "Completed" 
+    COMPLETED(3),
+    // This constant value indicates the internal value of the "Confirmed" 
+    CONFIRMED(4),
+    // This constant value indicates the internal value of the "Terminated" 
+    TERMINATED(5);
     
+    // Internal Variable
+    private int intValue = -1;
```

Comment:
I would prefer not to have these int values... I was thinking about calling "PROCEEDING"  like the previous constant "_PROCEEDING" so client dont need to change...